### PR TITLE
Expose service name and environment via MDC

### PR DESCRIPTION
**Pull request by seemk**
* New system property / env var: `SIGNALFX_ENVIRONMENT_NAME`. `default: ""`.
* If environment name is set, it is added to span tags as `environment`
* Expose both service name and environment to loggers similar to how trace and span IDs are exposed

### CODE REVIEW

**Review by iNikem — APPROVED** [review]

**Comment by seemk** [comment]
Note: changing the environment variable name to `SIGNALFX_ENV`.

**Review by johnbley — APPROVED** [review]

**Comment by johnbley** [comment]
LGTM, let me know when you want to turn a release and I can handle that in my Copious Spare Time.

**Comment by seemk** [comment]
Don't / won't merge yet - changing the configuration options a bit.

**Comment by seemk** [comment]
Replaced `SIGNALFX_ENVIRONMENT_NAME` with `SIGNALFX_ENV` to be more consistent with other agents. `ENV` already existed before, but only as an env var, so fixed this.

**Pull request merged**
